Update extract_images_sync to add sec_per_frame parameter

artineering opened the pull request:
Added sec_per_frame parameter to allow decimation of frames being synchronized and captured.
fixes #726

artineering committed:
Update extract_images_sync to add sec_per_frame parameter

mikeferguson commented:
Running this locally I hit: 
```
  File "/home/fergs/image/install/image_view/lib/image_view/extract_images_sync", line 97, in save
    if (self.sec_per_frame > 0) and (delay < self.sec_per_frame):
TypeError: '<' not supported between instances of 'Duration' and 'float'
```

mikeferguson commented on image_view/scripts/extract_images_sync:
```diff
@@ -51,10 +51,19 @@ class ExtractImagesSync(Node):
+
+        # Limit the throughput to 10fps by default.
+        self.sec_per_frame = self.declare_parameter(
+            'sec_per_frame', 0.1).value
```
we should probably not change default behavior - so this should probably be set to 0 by default

mikeferguson commented on image_view/scripts/extract_images_sync:
```diff
@@ -51,10 +51,19 @@ class ExtractImagesSync(Node):
         super().__init__('extract_images_sync')
         self.get_logger().info('Extract_Images_Sync Node has been started')
         self.seq = 0
+        self.time = self.get_clock().now()
```
let's use a more descriptive name than "time" - something like "last_frame" or "last_frame_time", etc

mikeferguson commented on image_view/scripts/extract_images_sync:
```diff
@@ -80,6 +89,17 @@ Typical command-line usage:
+        # Decimation is enabled if it is set to greater than
+        # zero. If it is enabled, exit if delay is less than
+        # the decimation rate specified.
+        if (self.sec_per_frame > 0) and (delay < self.sec_per_frame):
```
you can't directly compare a Duration and a float - need to do a conversion here:
```
  File "/home/fergs/image/install/image_view/lib/image_view/extract_images_sync", line 97, in save
    if (self.sec_per_frame > 0) and (delay < self.sec_per_frame):
TypeError: '<' not supported between instances of 'Duration' and 'float'
```

artineering committed:
Update extract_images_sync

1. Set default value for sec_per_frame to 0.
2. changed time variable to last_frame_time.

artineering committed:
Update extract_images_sync

Ensure delay is compared with the sec_per_frame value as a float.

mikeferguson commented on image_view/scripts/extract_images_sync:
```diff
@@ -51,10 +51,19 @@ class ExtractImagesSync(Node):
+
+        # Limit the throughput to 10fps by default.
+        self.sec_per_frame = self.declare_parameter(
+            'sec_per_frame', 0).value
```
default needs to be 0.0 so that the parameter is a float rather than an integer

mikeferguson committed:
make param float, proper comparison

mikeferguson commented:
I just pushed a commit that:

 * makes the param a float, rather than an integer
 * does proper conversion (there is no to_sec(), but we can make a Duration from secs_per_frame)

Tested locally and works
